conan/tools/cmake/toolchain/blocks.py

```diff
@@ -112,6 +112,53 @@ def context(self):
 
         return {"vs_runtimes": config_dict}
 
+class VSDebuggerEnvironment(Block):
+    template = textwrap.dedent("""
+        # Definition of CMAKE_VS_DEBUGGER_ENVIRONMENT
+        {% if vs_debugger_path %}
+        {% set genexpr = namespace(str='') %}
+        {% for config, value in vs_debugger_path.items() %}
+        {% set genexpr.str = genexpr.str +
+                                '$<$<CONFIG:' + config + '>:' + value|string + '>' %}
+        {% endfor %}
+        set(CMAKE_VS_DEBUGGER_ENVIRONMENT "PATH={{ genexpr.str }};%PATH%")                      
+        {% endif %}
+        """)
+    
+    def context(self):
+        os_ = self._conanfile.settings.get_safe("os")
+        build_type = self._conanfile.settings.get_safe("build_type")
+
+        if (os_ and not "Windows" in os_) or not build_type:
+            return None
+        
+        generator = self._conanfile.conf.get("tools.cmake.cmaketoolchain:generator", default="Visual Studio")
+        if not "Visual Studio" in generator:
+            return None
+        
+        config_dict = {}
+        if os.path.exists(CONAN_TOOLCHAIN_FILENAME):
+            existing_include = load(CONAN_TOOLCHAIN_FILENAME)
+            vs_debugger_environment = re.search(r"set\(CMAKE_VS_DEBUGGER_ENVIRONMENT \"PATH=([^)]*)>;%PATH%\"\)",
+                                           existing_include)
+            if vs_debugger_environment:
+                capture = vs_debugger_environment.group(1)
+                matches = re.findall(r"\$<\$<CONFIG:([A-Za-z]*)>:(.*)", capture)
+                config_dict = dict(matches)
```

Comment:
This is the part we might want to refactor, but that might be done later in another PR

Reply:
Removed the string generation from the jinja, and into python, although still very tricky to get right with the regexes. Fixed some issues when we have more than 2 configs, so it should be more robust.

IN a later PR I have some ideas for how to support multiconfig in a more straightforward way